app/src/main/java/com/example/pygmyhippo/database/ImageStorage.java

```diff
@@ -36,26 +30,28 @@ public class ImageStorage extends DBHandler {
      * @param listener
      * @author Griffin
      */
-    public void uploadImageToFirebase(Uri imageUri, DBOnCompleteListener<Event> listener) {
+    public void uploadImageToFirebase(Uri imageUri, StorageOnCompleteListener<Image> listener) {
         StorageReference storageRef = storage.getReference();
         String imageName = "events/" + UUID.randomUUID().toString();
         StorageReference imageRef = storageRef.child(imageName);
-        ArrayList<Event> events = new ArrayList<>();
+        ArrayList<Image> images = new ArrayList<>();
 
         imageRef.putFile(imageUri)
                 .addOnCompleteListener(task -> {
                     if (task.isSuccessful()) {
                         imageRef.getDownloadUrl().addOnSuccessListener(uri -> {
                             Log.d("FirebaseStorage", "Image uploaded successfully. URL: " +  uri.toString());
-                            Event newEvent = new Event();
-                            newEvent.setEventPoster(uri.toString());
-                            events.add(newEvent);
-                            listener.OnComplete(events, 3, DBOnCompleteFlags.SUCCESS.value);
+                            Image newImage = new Image();
+                            newImage.setUrl(uri.toString());
+                            newImage.setType(Image.ImageType.Event);
+
```

Comment:
Is storing the image still relating it to its event?

Reply:
we don't have an image object in the event, I only relate the image in the database with the Url. I'm using the image class as a way of passing and image url to the event. I'm not sure if this is the best approach.

Reply:
Should we consider any changes for this then? I mean if it works I'm okay with it though